packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -160,6 +170,11 @@ InnerBlocks = compose( [
 	} ),
 ] )( InnerBlocks );
 
+// Expose default appender placeholders as components.
+InnerBlocks.DefaultBlockAppender = DefaultBlockAppender;
+InnerBlocks.ButtonBlockAppender = ButtonBlockAppender;
+InnerBlocks.HideWhenChildBlocks = HideWhenChildBlocks;
```

Comment:
The user may still want to render an appender when child blocks exist (e.g.: a simpler version). Maybe we can provide a fallBack prop as we do in https://github.com/WordPress/gutenberg//blob/2e6199df033ca8b39ed89221f9b62c820ac88b0a/packages/block-editor/src/components/media-upload/check.js#L6.

But in that case maybe the name should be WhenEmpty or WhenNoBlockExists.

Reply:
Apologies @jorgefilipecosta I wasn't able to follow this. Can you explain?

Reply:
I think what @jorgefilipecosta means is that it might be useful to be able to render one version of the appender when there are no child blocks, and then a different version when there are child blocks. I can see advantages of that—as Jorge implies the first might have a call to action, then later after blocks are added it becomes a simpler `+` button.

I wonder if it might be worth splitting this off into a separate PR, as I don't see it as a critical piece and it seems like it needs some further exploration.

My goal would be adding the button appender to the `Group` block as soon as possible, and that block doesn't need this.